grails-app/controllers/au/org/ala/ecodata/DataSetSummaryController.groovy

```diff
@@ -27,6 +27,33 @@ class DataSetSummaryController {
         respond projectService.updateDataSet(projectId, dataSet)
     }
 
+    /**
+     * Updates multiple data sets for a project.
+     * This endpoint exists to support the use case of associating multiple data sets with a
+     * report and updating their publicationStatus when the report is submitted/approved.
+     *
+     * This method expects the projectId to be supplied via the URL and the data sets to be supplied in the request
+     * body as a JSON object with key="dataSets" and value=List of data sets.
+     */
+    def bulkUpdate(String projectId) {
+        Map postBody =  request.JSON
+        List dataSets = postBody?.dataSets
+
+        if (!projectId) {
+            render status:  HttpStatus.SC_BAD_REQUEST, text: "projectId is required"
+            return
+        }
+
+        dataSets.each { Map dataSet ->
+            if (dataSet.projectId && dataSet.projectId != projectId) {
+                render status: HttpStatus.SC_BAD_REQUEST, text: "projectId must match the data set projectId"
+                return
```

Comment:
just wondering if this return will exit controller action or just the each closure?

Reply:
Thanks @temi, I've fixed this now.